ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java

```diff
@@ -12,27 +12,57 @@
  */
 package tech.pegasys.pantheon.ethereum.mainnet.precompiles.privacy;
 
+import static java.nio.charset.StandardCharsets.UTF_8;
+
 import tech.pegasys.pantheon.ethereum.core.Gas;
 import tech.pegasys.pantheon.ethereum.core.PrivacyParameters;
 import tech.pegasys.pantheon.ethereum.mainnet.AbstractPrecompiledContract;
 import tech.pegasys.pantheon.ethereum.vm.GasCalculator;
+import tech.pegasys.pantheon.orion.Orion;
+import tech.pegasys.pantheon.orion.types.ReceiveRequest;
+import tech.pegasys.pantheon.orion.types.ReceiveResponse;
 import tech.pegasys.pantheon.util.bytes.BytesValue;
 
+import java.io.IOException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
 public class PrivacyPrecompiledContract extends AbstractPrecompiledContract {
-  private PrivacyParameters privacyParameters;
+  private final Orion orion;
+  private final String orionPublicKey;
+
+  private static final Logger LOG = LogManager.getLogger();
 
   public PrivacyPrecompiledContract(
       final GasCalculator gasCalculator, final PrivacyParameters privacyParameters) {
+    this(gasCalculator, privacyParameters.getPublicKey(), new Orion(privacyParameters.getUrl()));
+  }
+
+  PrivacyPrecompiledContract(
+      final GasCalculator gasCalculator, final String publicKey, final Orion orion) {
     super("Privacy", gasCalculator);
+    this.orion = orion;
+    this.orionPublicKey = publicKey;
   }
 
   @Override
   public Gas gasRequirement(final BytesValue input) {
-    return null; // We need to write the contract first.
+    return Gas.of(40_000L); // Not sure
   }
 
   @Override
   public BytesValue compute(final BytesValue input) {
+    try {
+      String key = new String(input.extractArray(), UTF_8);
+      ReceiveRequest receiveRequest = new ReceiveRequest(key, orionPublicKey);
+      ReceiveResponse receiveResponse = orion.receive(receiveRequest);
+      LOG.info("Got the response as ", receiveResponse.getPayload());
+      return BytesValue.wrap(receiveResponse.getPayload());
+      // pass it to private tx processor
```

Comment:
Did we forget to implement something or is this comment here as a reminder to do something in the future?

Reply:
Its a reminder to implement in the future.